docs/api/categories.md

```diff
@@ -0,0 +1,265 @@
+# Cagegories
+
+Categorize Speech Challenges or categories.
+
+## List all top level categories
+
+List all top level categories which do not have a parent category.
+
+### URL
+
+```http
+GET /categories HTTP/1.1
+```
+
+### Request
+
+```http
+GET /categories HTTP/1.1
+Accept: application/json
+```
+
+### Response
+
+The resonse is a JSON list with categories.
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+[
+  {
+    "id": "category_1",
+    "created": "2017-01-12T09:35:20Z",
+    "updated": "2017-01-12T09:35:20Z",
+    "name": "Category 1",
+    "description": "Some awesome description.",
+    "color": "#00f",
+    "imageUrl": "https://api.itslanguage.nl/download/UKbsMpBsXaJUsBbK",
+    "iconUrl": "https://api.itslanguage.nl/download/GdExSbs-ZVNnQUUe",
+    "categories": ["category_1_1", "category_1_2"],
```

Comment:
subcategories would be more suitable, however, I would not like to include this field. I rather query /categories again with extra filter: parent_category=category_1. Why? Because adding a new category in the database would require updating the parent category as well as adding the new category. Or extra queries need to be done in python at the time a user GETs /categories. Alternatively, like cgen has the API: we could provide the link where to GET the subcategories (HATEOAS). Then again, I like consistency and we don't provide links to other API calls in the output, except for download links.